servicediscovery/mdns/mdns.go

```diff
@@ -0,0 +1,71 @@
+// ------------------------------------------------------------
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+// ------------------------------------------------------------
+
+package mdns
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/dapr/components-contrib/servicediscovery"
+	"github.com/grandcat/zeroconf"
+)
+
+func NewMDNSResolver() *Resolver {
```

Comment:
Last comment, then lgtm:

It's better to return the `Resolver` interface here, and not the Resolver struct. You can make the Resolver struct unexported (private) by changing it to `resolver`.